packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx

```diff
@@ -58,24 +56,17 @@ const AccountList = forwardRef<HTMLDivElement, Props>(({ className, onSelect },
     const [selectedAccount, setSelectedAccount] = useAtom(selectedAccountAtom);
     const nav = useNavigate();
     const { t } = useTranslation('mainLayout');
-    const { jsonRpcUrl } = useAtomValue(networkConfigurationAtom);
     const identityNames = useAtomValue(identityNamesAtom);
-
     const [totalBalanceMap, setTotalBalanceMap] = useState<Map<string, bigint>>(new Map());
+    const accountInfos = accounts.map(useAccountInfo);
 
     useEffect(() => {
-        const emitter = new AccountInfoEmitter(jsonRpcUrl);
-        emitter.on('totalchanged', (accountInfo: AccountInfo, address: string) => {
-            setTotalBalanceMap(new Map(totalBalanceMap.set(address, accountInfo.accountAmount)));
-        });
-        // TODO Log the error instead of gobbling it.
-        emitter.on('error', noOp);
-        emitter.listen(accounts.map((account) => account.address));
-        return () => {
-            emitter.removeAllListeners('totalchanged');
-            emitter.stop();
-        };
-    }, [JSON.stringify(accounts)]);
+        for (const info of accountInfos) {
+            if (info) {
+                setTotalBalanceMap(new Map(totalBalanceMap.set(info.accountAddress, info.accountAmount)));
+            }
+        }
+    }, [JSONBig.stringify(accountInfos)]);
```

Comment:
This could be refactored to only `setTotalBalanceMap` once. (I think that might be better :thinking:)